cishouseholds/pipeline/run.py

```diff
@@ -61,11 +62,15 @@ def run_from_config():
     with spark_description_set("adding run status"):
         add_run_status(run_id, "started")
     pipeline_error_count = None
+
     try:
+        config_copy = copy.deepcopy(config)
+        pipeline_stage_list = {stage_name: stage for stage_name, stage in config["stages"].items() if stage.pop("run")}
```

Comment:
Can we change this back, so that the running logic below determines which stages are going to be run?

Then can we validate all stages, regardless of whether they are going to be run?
`run` and `when` are special cases that you need to account for. `run` must be boolean, and you may be best to discuss with @slamothe99  how to validate their entry for `when`